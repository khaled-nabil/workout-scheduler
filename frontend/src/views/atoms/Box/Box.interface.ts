import {
  FlexboxProps,
  LayoutProps,
  SpaceProps,
  BorderProps,
  BoxShadowProps,
  ColorProps,
} from 'styled-system';

export interface BoxAtom
  extends FlexboxProps,
    SpaceProps,
    LayoutProps,
    BorderProps,
    BoxShadowProps,
    ColorProps {
  className?: string;
}
