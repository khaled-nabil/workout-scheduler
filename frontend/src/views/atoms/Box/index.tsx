import React, { FC } from 'react';
import styled from 'styled-components';
import { flex, flexbox, space, layout, color, border } from 'styled-system';
import { css } from '@styled-system/css';
import { BoxAtom } from './Box.interface';

const StyledBox: FC<BoxAtom> = styled('div')(
  css({
    maxWidth: '100%',
    height: 'auto',
    boxSizing: 'border-box',
  }),
  flex,
  flexbox,
  space,
  layout,
  color,
  border,
);

const Box: React.FC<BoxAtom> = ({ children, ...props }) => (
  <StyledBox {...props}>{children}</StyledBox>
);

export default Box;
