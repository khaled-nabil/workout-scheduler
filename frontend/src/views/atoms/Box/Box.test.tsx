import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Box from './';

describe('atom / Box', () => {
  it('Renders in containment with default props', () => {
    const component = shallow(
      <Box>
        <span>Contained Span</span>
      </Box>,
    );
    expect(component.text()).toEqual('Contained Span');
    expect(component.find('span').exists()).toBeTruthy();
    expect(toJson(component)).toMatchSnapshot();
  });

  it('Renders with custom values when props passed', () => {
    const component = shallow(
      <Box className="class-name" flex={1}>
        <span>Contained Span</span>
      </Box>,
    );

    expect(component.find('span').exists()).toBeTruthy();
    expect(component.prop('className').trim()).toContain('class-name');
    expect(component.prop('flex')).toEqual(1);
    expect(toJson(component)).toMatchSnapshot();
  });
});
