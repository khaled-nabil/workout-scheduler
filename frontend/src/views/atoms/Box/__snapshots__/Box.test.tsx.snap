// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`atom / Box Renders in containment with default props 1`] = `
<styled.div>
  <span>
    Contained Span
  </span>
</styled.div>
`;

exports[`atom / Box Renders with custom values when props passed 1`] = `
<styled.div
  className="class-name"
  flex={1}
>
  <span>
    Contained Span
  </span>
</styled.div>
`;
